src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package debug
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system/syslog"
+	"gopkg.in/yaml.v2"
+)
+
+// Use the same date specs as journalctl (see `man journalctl`).
+const timeHelpText = `Date specifications should be of the format "2012-10-30
+18:17:16". If the time part is omitted, "00:00:00" is assumed. If only the seconds
+component is omitted, ":00" is assumed. If the date component is omitted, the current
+day is assumed. Alternatively the strings "yesterday", "today", "tomorrow" are
+understood, which refer to 00:00:00 of the day before the current day, the current
+day, or the day after the current day, respectively.  "now" refers to the current
+time. Finally, relative times may be specified, prefixed with "-" or "+", referring to
+times before or after the current time, respectively.`
+
+type step func() error
+
+type stepParams struct {
+	fs            afero.Fs
+	m             *sync.Mutex
```

Comment:
This can just be a `sync.Mutex`, no need for a pointer

Reply:
Since this gets passed to several other functions, wouldn't that cause the lock to be copied?